src/entity-manager/EntityManager.ts

```diff
@@ -704,10 +704,11 @@ export class EntityManager {
         const overwriteColumns = metadata.columns.filter(
             (col) =>
                 !conflictColumns.includes(col) &&
-                entities.some(
+                (entities.some(
                     (entity) =>
                         typeof col.getEntityValue(entity) !== "undefined",
-                ),
+                ) ||
+                    col.isUpdateDate),
```

Comment:
When `skipUpdateIfNoValuesChanged` option is set, it shouldn't add update date column otherwise it will update even when nothing has changed.